### PR TITLE
Update sealso section in st_transform, close #1956

### LINKED ISSUE #1956: `st_transform` docs are outdated in relation to projections 'not supported by GDAL'
The documentation of `st_transform` states that 

> Projecting simple feature geometries to projections not supported by GDAL may be done by [st_transform_proj](https://rdrr.io/pkg/lwgeom/man/st_transform_proj.html), part of package lwgeom.

As discussed with @Nowosad it is not clear that this is still true. We have tried many of the projections listed here: https://proj.org/operations/projections/index.html

All of the ones we've tried work with `st_transform`. A simple fix to the docs, if this is indeed the case (I'm not 100% sure), would be to remove that line in the docs. If there are examples where only `st_transform_proj` it may be worth mentioning them.

Context is updating Geocomputation with R following reviewer comments:

https://github.com/Robinlovelace/geocompr/commit/32e89551c34e92058124a463dd6ee1e273021cf4

This seems to be an improvement, reducing the number of functions people need to remember for reprojecting sf objects, if I understand correctly.

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Nudge on this @edzer, I guess you didn't see this. I believe this is good but happy to re-work/close it if not right or no longer relevant. Cheers from Leeds!

**Comment by edzer** [comment]
I did, and left some comments - maybe you missed those?

**Comment by Robinlovelace** [comment]
> I did, and left some comments - maybe you missed those?

Aha, yes didn't get a notification. Taking a look now...

**Comment by Robinlovelace** [comment]
Update on this: I still see no comments. See image below.

My guess: you made comments line-by-line but did not submit the review, with either Comment, Approve or Reject. In fact this situation of disappearing reviews has happened to me and my colleagues: I thought I left a review and was very confused when colleagues did not see the comments.

That's just my guess. Happy to receive the comments via other means if GitHub not working in this instance...

<details>

image

**Comment by Robinlovelace** [comment]
Better 'no comments' image:

image

**Comment by Robinlovelace** [comment]
https://github.com/r-spatial/sf/issues/1956

**Comment by edzer** [comment]
Ah, thanks - yes, the "finish review" button is pretty well hidden!

**Comment by Robinlovelace** [comment]
Thanks for the review Edzer. I will rebase and make only the relevant changes.

File: man/tidyverse.Rd [hunk 1]
```diff
@@ -161,7 +161,7 @@ anti_join.sf(x, y, by = NULL, copy = FALSE, suffix = c(".x", ".y"), ...)
 
 \item{add}{see corresponding function in dplyr}
 
-\item{x}{A pair of data frames, data frame extensions (e.g. a tibble), or
+\item{x, y}{A pair of data frames, data frame extensions (e.g. a tibble), or
```
[the remark this answers is not in the document]
Comment (Robinlovelace) [reply man/tidyverse.Rd]: Yes, I didn't make those changes manually, Roxygen did.

File: man/vctrs.Rd [hunk 2]
```diff
@@ -23,9 +23,7 @@ vec_cast.sfc(x, to, ...)
 \method{vec_cast.sfc}{default}(x, to, ...)
 }
 \arguments{
-\item{x}{Vector types.}
-
-\item{y}{Vector types.}
+\item{x, y}{Vector types.}
```
[the remark this answers is not in the document]
Comment (Robinlovelace) [reply man/vctrs.Rd]: I haven't re-documented anything in the latest PR.

File: sf.Rproj [hunk 3]
```diff
@@ -13,7 +13,6 @@ RnwWeave: knitr
 LaTeX: XeLaTeX
 
 AutoAppendNewline: Yes
-StripTrailingWhitespace: No
```
[the remark this answers is not in the document]
Comment (Robinlovelace) [reply sf.Rproj]: Ask RStudio, wasn't me. I've left it as is now.

**Comment by Robinlovelace** [comment]
Heads-up @edzer after a few more circles I think this is ready to go :rocket: (hopefully more successful than the recent planned NASA launch)

**Comment by Robinlovelace** [comment]
Another nudge on this @edzer. It's not a lot of commits for a minor change, let me know if you want a cleaner commit history and I can submit a fresh PR.

**Comment by edzer** [comment]
thanks!

**Pull request merged**